mpp/include/mpp_io_read.inc

```diff
@@ -27,159 +27,267 @@
-#define READ_RECORD_CORE_ read_record_core
+#define READ_RECORD_CORE_ read_record_core_r8
 #undef READ_RECORD_
-#define READ_RECORD_ read_record
-#define MPP_READ_2DDECOMP_2D_ mpp_read_2ddecomp_r2d
+#define READ_RECORD_ read_record_r8
+#define MPP_READ_2DDECOMP_2D_ mpp_read_2ddecomp_r2d_r8
 #undef MPP_READ_2DDECOMP_3D_
-#define MPP_READ_2DDECOMP_3D_ mpp_read_2ddecomp_r3d
+#define MPP_READ_2DDECOMP_3D_ mpp_read_2ddecomp_r3d_r8
 #undef MPP_READ_2DDECOMP_4D_
-#define MPP_READ_2DDECOMP_4D_ mpp_read_2ddecomp_r4d
+#define MPP_READ_2DDECOMP_4D_ mpp_read_2ddecomp_r4d_r8
 #undef MPP_TYPE_
-#define MPP_TYPE_ real
+#define MPP_TYPE_ real(KIND=r8_kind)
 #include <mpp_read_2Ddecomp.h>
 
-#ifdef OVERLOAD_R8
 #undef READ_RECORD_CORE_
-#define READ_RECORD_CORE_ read_record_core_r8
+#define READ_RECORD_CORE_ read_record_core_r4
 #undef READ_RECORD_
-#define READ_RECORD_ read_record_r8
+#define READ_RECORD_ read_record_r4
 #undef MPP_READ_2DDECOMP_2D_
-#define MPP_READ_2DDECOMP_2D_ mpp_read_2ddecomp_r2d_r8
+#define MPP_READ_2DDECOMP_2D_ mpp_read_2ddecomp_r2d_r4
 #undef MPP_READ_2DDECOMP_3D_
-#define MPP_READ_2DDECOMP_3D_ mpp_read_2ddecomp_r3d_r8
+#define MPP_READ_2DDECOMP_3D_ mpp_read_2ddecomp_r3d_r4
 #undef MPP_READ_2DDECOMP_4D_
-#define MPP_READ_2DDECOMP_4D_ mpp_read_2ddecomp_r4d_r8
+#define MPP_READ_2DDECOMP_4D_ mpp_read_2ddecomp_r4d_r4
 #undef MPP_TYPE_
-#define MPP_TYPE_ real(DOUBLE_KIND)
+#define MPP_TYPE_ real(KIND=r4_kind)
 #include <mpp_read_2Ddecomp.h>
-#endif
 
+#undef READ_RECORD_
+#define READ_RECORD_ read_record_r8
+#undef MPP_READ_COMPRESSED_1D_
+#define MPP_READ_COMPRESSED_1D_ mpp_read_compressed_r1d_r8
+#undef MPP_READ_COMPRESSED_2D_
+#define MPP_READ_COMPRESSED_2D_ mpp_read_compressed_r2d_r8
+#undef MPP_READ_COMPRESSED_3D_
+#define MPP_READ_COMPRESSED_3D_ mpp_read_compressed_r3d_r8
+#undef MPP_TYPE_
+#define MPP_TYPE_ real(KIND=r8_kind)
+#include <mpp_read_compressed.h>
+
+#undef READ_RECORD_
+#define READ_RECORD_ read_record_r4
 #undef MPP_READ_COMPRESSED_1D_
-#define MPP_READ_COMPRESSED_1D_ mpp_read_compressed_r1d
+#define MPP_READ_COMPRESSED_1D_ mpp_read_compressed_r1d_r4
 #undef MPP_READ_COMPRESSED_2D_
-#define MPP_READ_COMPRESSED_2D_ mpp_read_compressed_r2d
+#define MPP_READ_COMPRESSED_2D_ mpp_read_compressed_r2d_r4
 #undef MPP_READ_COMPRESSED_3D_
-#define MPP_READ_COMPRESSED_3D_ mpp_read_compressed_r3d
+#define MPP_READ_COMPRESSED_3D_ mpp_read_compressed_r3d_r4
 #undef MPP_TYPE_
-#define MPP_TYPE_ real
+#define MPP_TYPE_ real(KIND=r4_kind)
 #include <mpp_read_compressed.h>
 
 #include <mpp_read_distributed_ascii.inc>
 
-! <SUBROUTINE NAME="mpp_read_r4D" INTERFACE="mpp_read">
+! <SUBROUTINE NAME="mpp_read_r4D_r8" INTERFACE="mpp_read">
+!   <IN NAME="unit" TYPE="integer"></IN>
+!   <IN NAME="field" TYPE="type(fieldtype)"></IN>
+!   <INOUT NAME="data" TYPE="real(KIND=r8_kind)" DIM="(:,:,:,:)"></INOUT>
+!   <IN NAME="tindex" TYPE="integer"></IN>
+! </SUBROUTINE>
+    subroutine mpp_read_r4D_r8( unit, field, data, tindex)
```

Comment:
Should probably be converted to doxygen here and below.

Reply:
This can be ignored.  I think these routines are being deprecated and slated for removal.